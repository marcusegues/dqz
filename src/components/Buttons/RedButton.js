import React from 'react';
import { ScrollView, Text, View } from 'react-native';
import Touchable from 'react-native-platform-touchable';
import * as colors from './../../styles/colors';

const RedButton = ({ text, onPress, style }) => (
  <View>
    {/*<View style={{ margin: 16 }}>*/}
    <Touchable
      onPress={onPress}
      style={{ backgroundColor: colors.MAIN_RED, ...style }}
      background={Touchable.Ripple(colors.MAIN_RED)}
    >
      <Text style={{ textAlign: 'center', color: '#fff' }}>{text}</Text>
    </Touchable>
  </View>
);
export default RedButton;
