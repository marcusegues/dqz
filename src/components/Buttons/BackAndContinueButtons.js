// @flow
import React from 'react';
import type { ComponentType } from 'react';
// $FlowFixMe
import { View } from 'react-native';
import { translate } from 'react-i18next';
import { verticalScale } from '../../styles/Scaling';
import { GREY } from '../../styles/colors';
import { RedButton } from './RedButton';
import type { TFunction } from '../../types/generalTypes';

const ownStyles = {
  redButtonContainerWrapper: {
    width: '95%',
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignSelf: 'center',
    marginBottom: verticalScale(16),
  },
  redButtonWrapper: {
    flex: 0.47,
  },
};

type BackAndContinueButtonsProps = {
  onPressBack: () => void,
  onPressContinue: () => void,
  textBack?: string,
  textContinue?: string,
  backDisabled?: boolean,
  continueDisabled?: boolean,
};

const BackAndContinueButtonsInner = ({
  onPressBack,
  onPressContinue,
  textBack,
  textContinue,
  backDisabled = false,
  continueDisabled = false,
  t,
}: BackAndContinueButtonsProps & { t: TFunction }) => (
  <View style={ownStyles.redButtonContainerWrapper}>
    <View style={ownStyles.redButtonWrapper}>
      <RedButton
        onPress={() => onPressBack()}
        text={textBack || t('buttonBack')}
        buttonStyle={{ backgroundColor: GREY }}
        confirmationDisabled={backDisabled}
      />
    </View>
    <View style={ownStyles.redButtonWrapper}>
      <RedButton
        onPress={() => onPressContinue()}
        text={textContinue || t('buttonContinue')}
        confirmationDisabled={continueDisabled}
      />
    </View>
  </View>
);

export const BackAndContinueButtons = (translate(['general'])(
  BackAndContinueButtonsInner
): ComponentType<BackAndContinueButtonsProps>);
