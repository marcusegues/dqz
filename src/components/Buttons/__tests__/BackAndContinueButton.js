import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import { BackAndContinueButtons } from '../BackAndContinueButtons';
import { RedButton } from '../RedButton';

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

describe('BackAndContinueButton', () => {
  test('renders according to snapshot', () => {
    const component = renderer
      .create(
        <BackAndContinueButtons
          text="text"
          onPressBack={jest.fn()}
          onPressContinue={jest.fn()}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('we can click first button', () => {
    const clickFn = jest.fn();
    const component = shallow(
      <BackAndContinueButtons
        text="text"
        onPressBack={clickFn}
        onPressContinue={jest.fn()}
      />
    ).dive();
    component.find(RedButton).forEach((child, i) => {
      if (i === 0) {
        child.simulate('press');
      }
    });
    expect(clickFn).toBeCalled();
  });

  test('we can click second button', () => {
    const clickFn = jest.fn();
    const component = shallow(
      <BackAndContinueButtons
        text="text"
        onPressBack={jest.fn()}
        onPressContinue={clickFn}
      />
    ).dive();
    component.find(RedButton).forEach((child, i) => {
      if (i === 1) {
        child.simulate('press');
      }
    });
    expect(clickFn).toBeCalled();
  });
});
