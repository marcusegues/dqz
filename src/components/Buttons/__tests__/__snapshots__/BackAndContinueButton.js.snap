// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackAndContinueButton renders according to snapshot 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "marginBottom": 31.388235294117646,
      "width": "95%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.47,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignSelf": "center",
          "backgroundColor": "#DC0018",
          "borderColor": "#ddd",
          "borderRadius": 3,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 31.388235294117646,
          "width": "95%",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#9B9B9B",
            "borderRadius": 3,
            "flex": 1,
            "opacity": 1,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 1,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 3,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#fff",
              "fontFamily": "roboto_medium",
              "fontSize": 22,
              "fontWeight": "500",
              "paddingVertical": 31.388235294117646,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 0.47,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignSelf": "center",
          "backgroundColor": "#DC0018",
          "borderColor": "#ddd",
          "borderRadius": 3,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 31.388235294117646,
          "width": "95%",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#DC0018",
            "borderRadius": 3,
            "flex": 1,
            "opacity": 1,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 1,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 3,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#fff",
              "fontFamily": "roboto_medium",
              "fontSize": 22,
              "fontWeight": "500",
              "paddingVertical": 31.388235294117646,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
